### PR TITLE
Feat: WandbCallback upgrades

### PR DESCRIPTION
**Pull request by parambharat**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/integrations/integration_utils.py [hunk 1]
```diff
@@ -751,6 +754,10 @@ def setup(self, args, state, model, **kwargs):
                 self._wandb.watch(model, log=_watch_model, log_freq=max(100, state.logging_steps))
             self._wandb.run._label(code="transformers_trainer")
 
+            # add number of model parameters to wandb config
+            if isinstance(model, (PreTrainedModel, TFPreTrainedModel)):
+                self._wandb.config["model/num_parameters"] = model.num_parameters()
```
Comment (morganmcg1) [inline src/transformers/integrations/integration_utils.py:799]: Does this have to be logged both on `setup` and `on_train_begin` and 'on_save`?

Comment (parambharat) [reply src/transformers/integrations/integration_utils.py:799]: In `setup` we log it to the config, in `on_train_begin` and `on_save` we log it to the model artifact metadata and the checkpoint metadata respectively. This way, the run config, and metadata for the model and checkpoint metadata will all have the model parameter making it easier for reproducibility. That was my thought process here.

**Review by Dezzj — COMMENTED** [review]
Veiwed

File: src/transformers/integrations/integration_utils.py (hunk 1 above)
Comment (morganmcg1) [reply src/transformers/integrations/integration_utils.py:799]: makes sense, thanks!

**Pull request closed without merging**